game/src/core/utils/files/FileSystemUtil.java

```diff
@@ -0,0 +1,134 @@
+package core.utils.files;
+
+import core.utils.components.path.IPath;
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.util.Collections;
+import java.util.Objects;
+
+/** This class contains utility methods for working with files and directories. */
+public class FileSystemUtil {
+  // This instance constant can be used to determine from where the game was started.
+  @SuppressWarnings("InstantiationOfUtilityClass")
+  private static final Object HELPER_INST = new FileSystemUtil();
```

Comment:
@tgrothe Das finde ich nicht schön gelöst. Du brauchst an zwei oder drei Stellen ein Objekt - warum erzeugst Du es Dir dann nicht dort vor Ort? So musst Du eine Warning unterdrücken, was wirklich unschön ist.

Reply:
@cagix hmm, aber dann müssen wir für jeden aufruf von search oder get ein neues objekt anlegen... wirklich? dann ändere ich das. :)

Reply:
wenn ich einen get- oder search-aufrufe habe - dann passiert das doch auf einem existierenden objekt. kannst du nicht das dann auch für die getClass()-abfrage nutzen? ich verstehe noch nicht wirklich, warum wir hier ein neues objekt brauchen?

Reply:
> wenn ich einen get- oder search-aufrufe habe - dann passiert das doch auf einem existierenden objekt. kannst du nicht das dann auch für die getClass()-abfrage nutzen? ich verstehe noch nicht wirklich, warum wir hier ein neues objekt brauchen?

es muss ein objekt von einer eigenen klasse erstellt werden (also eine klasse, die in unserer codebase ist), sonst funktioniert es nicht. das objekt darf dabei nicht via reflection erstellt werden.

ich ändere das gerade...